project-reports/2023/2023-Q2-Hyperledger-Iroha.md

```diff
@@ -0,0 +1,97 @@
+---
+layout: default
+title: 2023 Q2 Hyperledger Iroha
+parent: 2023
+grand_parent: Project Updates
+---
+
+# Project Health
+
+Currently, active maintenance of Iroha 1 has stopped and there are no plans for major releases. However, discussions to release the current RC continue because it will require more than one team switching from everyday tasks to focus on this.
+
+All Iroha 2 developers are now fully onboarded and at maximum productivity.
+
+Since development of the Ursa project has been discontinued, the Iroha team is rewriting its essential parts to integrate in the Iroha codebase. In addition, thanks to Arun S M, the Iroha 2 BCE team will start collaborating with the Hyperledger Blockchain Explorer team.
+
+# Questions/Issues for the TSC
+
+None.
+
+# Releases
+
+Past quarter releases:
+
+* v2.0.0-pre-rc.13
+* v2.0.0-pre-rc.15
+
+# Overall Activity in the Past Quarter
+
+According to the [Iroha Activity Dashboard](https://insights.lfx.linuxfoundation.org/projects/hyperledger%2Firoha/dashboard;subTab=technical?time=%7B%22from%22:%222023-01-01T07:00:00.000Z%22,%22type%22:%22absolute%22,%22to%22:%222023-03-31T07:00:00.000Z%22%7D) for the first quarter of 2023 (January-March), the Iroha repo has 129 PRs submitted by 28 contributors.
+
+## Iroha v1
+
+Documentation restoration is in progress.
+
+Grzegorz Bazior started an Iroha 1 mentorship: "[extend queries with optional arguments](https://mentorship.lfx.linuxfoundation.org/project/06a25b14-a56e-4294-a89d-05f1dc74106c)".
+
+## Iroha v2
+
+### Code updates
+
+Builtin validators were replaced with runtime validators. Now it is possible to plug in any kind of custom user-defined validation.
+
+The `WASM` smartcontracts now have a logging feature, which is intended to help with debugging. ([#3264](https://github.com/hyperledger/iroha/pull/3264))
+
+A new pipeline event has been added: "transaction expired" ([#3010](https://github.com/hyperledger/iroha/issues/3010), [`067842`](https://github.com/hyperledger/iroha/commit/06784290b192d2d942426763874d4cc49d0485f0)) which improves transaction processing under a high load.
+
+Topology rotation was rewritten:
+- Previously, it was possible for the leader to manipulate the block and affect the topology in the next consensus round
+- Topology is not calculated based on hash of the current block anymore, it's determined from the previous topology and the view change index
+
+The `iroha_client_cli` output is now machine readable, leading to improved tests.
```

Comment:
Additionally an SDK based off of using `iroha_client_cli` and a common JSON API was proposed, and will be merged.